### ISSUE TITLE
Any plans for converting trained tensorflow models to matconvnet models?

### ISSUE THREAD

**Issue by Naruto-Sasuke** [vlfeat/matconvnet#773]

**Comment by albanie** [comment]
This is a good idea - I'll look into it.

**Comment by Usmankhalid123** [comment]
any progress,..

**Comment by idofr** [comment]
Should the weights be transferable? That is, given two parallel implementations, could I just copy the weights?

**Comment by albanie** [comment]
The fundamental issue is that Tensorflow is defined to work at a different level of abstraction (the "op") compared to frameworks like matconvnet which group computation into layers.  For this reason, writing importers from caffe and [pytorch](https://github.com/albanie/mcnPyTorch)  is comparatively "simple", but Tensorflow is considerably more tricky (you have to essentially invert the mapping between ops and layers by pattern matching). I did try putting something together, but it's extremely error prone/user-unfriendly so it stayed offline.

**Comment by idofr** [comment]
That's the reason i was asking about the weights directly instead of a complete translation.

**Comment by albanie** [comment]
It's certainly possible if you don't mind poking around a bit.  I've added my working progress to a public repo [here](https://github.com/albanie/mcnTensorflow) in case it's useful as a starting point.

**Comment by idofr** [comment]
Cheers, I'll have a look in the next few days but I gave it a few hours today and got everything to import properly. The only problem was the althought the weights were correct the accuracies were not much better than random. I tried it on a small MNIST example with conv and batch norm

**Comment by idofr** [comment]
Alright, so I've had a deeper look into my porting. When I only use conv layers, everything is find. I get slightly worse accuracies on MNIST but it's nothing too significant. However with every batch norm layer I add, the accracy decreases significantly. Deeper layers have a much stronger affect.
